sklearn/metrics/cluster/_dbcv_internals.py

```diff
@@ -0,0 +1,329 @@
+# Authors: The scikit-learn developers
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import warnings
+
+import numpy as np
+
+from ..pairwise import pairwise_distances
+
+
+def _density_separation(
+    X,
+    labels,
+    cluster_id1,
+    cluster_id2,
+    internal_nodes1,
+    internal_nodes2,
+    core_distances1,
+    core_distances2,
+    metric="euclidean",
+    no_coredist=False,
```

Comment:
For this particular function we may be able to ignore the `no_coredist` parameter and instead we can simply check if either of the core_distances is None. Actually I think it might suffice to check `if core_distances1 is None` and we can update the docstrings for both these parameters to reflect this. What do you think?
```suggestion
    core_distances1=None,
    core_distances2=None,
    metric="euclidean",
```

Reply:
Tbh I don't particularly like that idea. I get where you're coming from but I like the handling of core distance toggling to be more explicit. This way if, in some theoretical circumstance, possibly in the future, the core distance args are `None` by mistake, it will actually cause an error instead of silently switching off core distances.

It also makes for a function signature that's more similar/consistent with the other, previously mentioned function.

Let me know if you agree. If you don't insist, I'd prefer to keep it this way.

Reply:
Yes I think it is fine to keep it the way is is currently.